web-ng/src/app/components/layer-dialog/layer-dialog.component.ts

```diff
@@ -165,12 +166,13 @@ export class LayerDialogComponent implements OnDestroy {
     if (emptyFields.size) {
       return;
     }
+    const layerCount = await this.getLayerCount();
     const fields = this.layerService.convertFieldsListToMap(this.fields);
     const formId = this.form?.id;
     const forms = this.layerService.createForm(formId, fields);
     const layer = new Layer(
       this.layer?.id || '',
-      /* index */ this.layerCount,
+      /* index */ layerCount,
```

Comment:
I actually meant that we should set this value outside the component, in this case in the LayerService. That way there's no need to add a dependency on the active project.

Side nit: Please update the name `updateLayer` to `addOrUpdateLayer` if appropriate.

Reply:
Thanks!